### PR TITLE
Kubernetes source plugin

### PR DESCRIPTION
**Pull request by huseyinbabal**
## Description

Changes proposed in this pull request:

- Restructured & rewrote kubernetes related documentations
- Removed filters documentation since it is part of Kubernetes plugin now.

### CODE REVIEW

File: docs/configuration/source/kubernetes.md [hunk 1]
```diff
@@ -177,132 +177,50 @@ event:
...
-    labelsSet: true # set to `true` in `first-source`, other sources didn't specify its value
-  ingress:
-    backendServiceValid: true # set in `first-source` and `second-source` to `false`, overriden in `third-source` to `true`
-    tlsSecretValid: true # set in `second-source`
+# Filter settings for various sources.
+# Currently, all filters are globally enabled or disabled.
+# You can enable or disable filters with `@Botkube filters` commands.
+filters:
+  kubernetes:
+    # If true, enables support for `botkube.io/disable` and `botkube.io/channel` resource annotations.
```
Comment (mszostok) [inline docs/configuration/source/kubernetes.md:220]:
```suggestion
    # If true, enables support for `botkube.io/disable` resource annotations.
```

File: docs/configuration/source/kubernetes.md [hunk 2]
```diff
@@ -319,280 +237,323 @@ sources:
...
+              "services",
+              "ingresses",
+            ]
+        # -- Filter settings for various sources.
+        # Currently, all filters are globally enabled or disabled.
+        # You can enable or disable filters with `@Botkube enable/disable filters` commands.
+        # @default -- See the `values.yaml` file for full object.
+        filters:
+          kubernetes:
+            # -- If true, enables support for `botkube.io/disable` and `botkube.io/channel` resource annotations.
```
Comment (mszostok) [inline docs/configuration/source/kubernetes.md:449]:
```suggestion
            # -- If true, enables support for `botkube.io/disable` resource annotations.
```

**Review by mszostok — COMMENTED** [review]
Please also update `custom-source.md` file to reflect new syntax with `api.Message`. Thanks!

**Comment by huseyinbabal** [comment]
@mszostok addressed the comments thanks

File: docs/plugin/custom-source.md [hunk 3]
```diff
@@ -117,7 +117,9 @@ For a final implementation, see the [Botkube template repository](./quick-start.
    			case <-ctx.Done():
    				ticker.Stop()
    			case <-ticker.C:
-   				out.Output <- []byte("Ticker Event")
+   				out.Event <- source.Event{
```
Comment (mszostok) [inline docs/plugin/custom-source.md:120]: Please fix the indention here. Thanks 🙂 
Screenshot 2023-03-08 at 15 49 05

**Review by mszostok — APPROVED** [review]
Good job 🚀 One minor comment 👍

File: docs/plugin/custom-source.md (hunk 3 above)
Comment (huseyinbabal) [reply docs/plugin/custom-source.md:120]: Screen Shot 2023-03-09 at 10 56 43

**Pull request merged**